lib/improver/tests/utilities/test_temporal.py

```diff
@@ -549,34 +549,39 @@ def setUp(self):
     def test_time_coord(self):
         """Test that the nearest time point within the time coordinate is
         extracted."""
+        # print(self.cube.coords('time'))
         expected = self.cube[:, 0, :, :]
-        time_point = datetime.datetime(2015, 11, 23, 6, 0)
-        result = extract_nearest_time_point(self.cube, time_point)
+        time_point = datetime.datetime(2015, 11, 23, 6, 31)
+        result = extract_nearest_time_point(self.cube, time_point,
+                                            allowed_dt_difference=1800)
         self.assertEqual(result, expected)
 
     def test_time_coord_lower_case(self):
         """Test that the nearest time point within the time coordinate is
         extracted, when a time of 07:30 is requested."""
         expected = self.cube[:, 0, :, :]
         time_point = datetime.datetime(2015, 11, 23, 7, 30)
-        result = extract_nearest_time_point(self.cube, time_point)
+        result = extract_nearest_time_point(self.cube, time_point,
+                                            allowed_dt_difference=1800)
         self.assertEqual(result, expected)
 
     def test_time_coord_upper_case(self):
         """Test that the nearest time point within the time coordinate is
         extracted, when a time of 07:31 is requested."""
         expected = self.cube[:, 1, :, :]
         time_point = datetime.datetime(2015, 11, 23, 7, 31)
-        result = extract_nearest_time_point(self.cube, time_point)
+        result = extract_nearest_time_point(self.cube, time_point,
+                                            allowed_dt_difference=1800)
         self.assertEqual(result, expected)
 
     def test_forecast_reference_time_coord(self):
         """Test that the nearest time point within the forecast_reference_time
         coordinate is extracted."""
         expected = self.cube
-        time_point = datetime.datetime(2015, 11, 23, 6, 0)
+        time_point = datetime.datetime(2015, 11, 23, 3, 29)
         result = extract_nearest_time_point(
-            self.cube, time_point, time_name="forecast_reference_time")
+            self.cube, time_point, time_name="forecast_reference_time",
+            allowed_dt_difference=1800)
         self.assertEqual(result, expected)
```

Comment:
I'm not sure this test ever did anything sensible. There is only one forecast reference time coordinate, so I'm not convinced this shows me anything other than the function returning the input (which is what I want, but is all it could ever do).

Adding another FRT to the cube would make this a better test.

```
        later_frt = self.cube.copy()
        later_frt.coord('forecast_reference_time').points = (
            later_frt.coord('forecast_reference_time').points + 3600)
        cubes = iris.cube.CubeList([self.cube, later_frt])
        cube = cubes.merge_cube()
        expected = self.cube
        time_point = datetime.datetime(2015, 11, 23, 3, 29)
        result = extract_nearest_time_point(
            cube, time_point, time_name="forecast_reference_time",
            allowed_dt_difference=1800)
        self.assertEqual(result, expected)
```

Reply:
Done